.github/workflows/marketplace-release.yaml

```diff
@@ -33,28 +41,84 @@ jobs:
           git submodule update --init --recursive plugins
           git submodule update --init --recursive tenzir
 
+      - name: Configure ssh-agent
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.TENZIR_PLUGINS_DEPLOY_KEY }}
+
+      - name: Update private submodules
+        run: |
+          git submodule update --init --recursive contrib/tenzir-plugins
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
+      # Login to ghcr before the build because we use it for caching.
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: tenzir-bot
+          password: ${{ secrets.TENZIR_BOT_GITHUB_TOKEN }}
+
+      # In theory it should be possible to assume this role directly, at least it's
+      # trust policy is permitting this. But somewhere along the stack there's a check
+      # that the OIDC provider is in the same AWS account as the role we're trying to
+      # assume, so we're using this two-step chain as a workaround.
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v4
         with:
           role-to-assume: arn:aws:iam::622024652768:role/ecr-tenzir-ce-github-access
-          aws-region: us-east-1
+          aws-region: eu-west-1
+          role-session-name: GithubActionsCISession
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::390844774768:role/aws-marketplace-github-access
+          aws-region: eu-west-1
+          role-session-name: GithubActionsCISession
+          role-skip-session-tagging: true
+          role-chaining: true
 
       - name: Login to AWS ECR
         uses: aws-actions/amazon-ecr-login@v2
+        env:
+          AWS_REGION: us-east-1
+        with:
+          registries: 709825985650
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          buildkitd-config-inline: |
+            [worker.oci]
+              max-parallelism = 1
 
       - name: Build and Push Marketplace Image
+        if: (github.event.inputs.existing_image == '' || github.event.inputs.existing_image == null)
         env:
+          # TODO: add multi-arch support
```

Comment:
We have this solved in the Docker workflow already, which also already uploads to multiple registries. I think you can just add the login step there, and add the registry to the list of registries we upload to. This should not repeat all the build logic.